bpf/lib/common.h

```diff
@@ -472,6 +472,7 @@ struct egress_gw_policy_entry6 {
 	union v6addr egress_ip;
 	__u32 gateway_ip;
 	__u32 reserved[3]; /* reserved for future extension, e.g. v6 gateway_ip */
+	__u32 egress_ifindex;
```

Comment:
Drive by comment: was this u32 variable one of the three u32 we reserved with the array above or do we still need 96 bits reserved for future use?

Reply:
I liked your idea of eventually turning those unused bytes into an IPv6 address (for IPv6-only configs).

Reply:
> In my initial PR it should have been 5x32 bits. 128 for the ipv6 gateway IP

I think we can follow Paul's approach in https://github.com/cilium/cilium/pull/38296, and unionize the IPv4 and IPv6 address field.

> and 32 for a future use-case you had in mind, which is presumably this one?

yep!

Reply:
>I think we can follow Paul's approach in https://github.com/cilium/cilium/pull/38296, and unionize the IPv4 and IPv6 address field.
Oh, that's neat. Then 3 reserved is still correct, phew.